lib/spack/spack/paths.py

```diff
@@ -50,7 +50,8 @@
 
 #: User configuration location
 user_config_path = os.path.expanduser('~/.spack')
-
+user_bootstrap_path = os.path.join(user_config_path, 'bootstrap')
+user_bootstrap_store = os.path.join(user_bootstrap_path, 'store')
```

Comment:
I think the bootstrap default location should be in the Spack tree, not in the config directory

Reply:
I thought about that, but it introduces redundancy. With the current location different instances of Spack would reuse the same bootstrapped software if it matches. If the store is in the Spack tree, each Spack instance will rebuild `clingo` (and the other dependencies when we'll add them later)

Reply:
Yes, but it also reduces redundancy in another dimension -- in a shared Spack instance, currently every users needs to individually bootstrap.